memcached/internal/memcached.c

```diff
@@ -208,6 +208,9 @@ memcached_loop(struct memcached_connection *con)
 			memcached_flush(con);
 		fiber_reschedule();
 		batch_count = 0;
+                if (ibuf_used(con->in) > 0) {
+                    goto next;
+                }
```

Comment:
Regarding code:

* The idea of the fix looks meaningful for me.
* Indented with whitespaces instead of tabs.

Regarding description:

* The fix does not look specific for a GET request: is it just for example? When we'll write a release notes, it would be good to offer information about possibly affected scenarios: should a user, which looks something similar with another command, try to update?
* Am I understand right that all SETQ, SETQ and GET requests lands to an input buffer before we start to process a first request? Or GET lands to the buffer while we processing SETQ commands? If it is correct, I would highlight this fact — it is key for understanding what is wrong.
* The 'what goes wrong' paragraph ends with 'we do a network read' and it left me with the question 'what is wrong here with reading from network?' Maybe it worth to clarify that the next command is already in the buffer and a client waits for a response before sending anything else (synchronous interation).

Thoughts for future:

* It would be nice to describe this code as a state diagram.
* It would be nice to have a unit testing that controls `read()` and fills buffer in different portions (several commands in a batch, partial command and so on).

Reply:
Bugfix has been moved to a separate branch, I'll create a separate PR and will answer there to comments.

Reply:
See a new PR https://github.com/tarantool/memcached/pull/94 and new comments with answers in issue #73.

> It would be nice to have a unit testing that controls read() and fills buffer in different portions (several commands in a batch, partial command and so on).

In general, I agree. In particular, I think we need to evaluate current code coverage made by existed regression test suite. Also I think that unit test is not appropriate type of test, because read-parse-process loop involves about 10 different functions and is not a pure function because uses `con` structure whose members mutated by other functions. But we definitely need to make this part of code more clear.

Reply:
Let's continue discussion in https://github.com/tarantool/memcached/pull/94.

Reply:
My concerns were resolved in PR #94, so we have nothing to discuss now.